### PR TITLE
Add documentation section on query optimization

### PR DESCRIPTION
**Pull request by kmshin1397**
Per @stefanv's suggestion, I put together a section in the docs on the general steps I take to do the various query optimizations I've been working on. Definitely open to suggestions on improvements/additions :)

### CODE REVIEW

File: doc/query_analysis.md [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+```
+
+would yield the following Postgres-dialect SQL query
+
+```SQL
+SELECT instruments.id FROM instruments WHERE instruments.id = 1;
+```
+
+A good first step to debugging or looking for inefficiencies in a query made by the SkyPortal backend
+is to look at the actual SQL generated by SQLAlchemy for the query to do so, you can do something like this:
```
Comment (acrellin) [inline doc/query_analysis.md:28]:
```suggestion
is to look at the actual SQL generated by SQLAlchemy for the query. To do so, you can do something like this:
```

File: doc/query_analysis.md [hunk 2]
```diff
@@ -0,0 +1,94 @@
...
+```
+
+Output:
+
+```
+x = :x_1
+x = %(x_1)s
+```
+
+Finally, you should note that in the SkyPortal API handlers, most of the SQLAlchemy `Query` objets
```
Comment (acrellin) [inline doc/query_analysis.md:58]:
```suggestion
Finally, you should note that in the SkyPortal API handlers, most of the SQLAlchemy `Query` objects
```

File: doc/query_analysis.md [hunk 3]
```diff
@@ -0,0 +1,94 @@
...
+that you would print out in this manner originate from our [permissions framework](permissions.html)
+through the `query_accessible_rows()` method. Thus, if you're ever lost looking for those
+`DBSession().query()...` lines to do this sort of print-debugging, you may want to look for the
+`query_accessible_rows()` lines instead. Similarly, if you're ever wondering where certain parts of
+the printed out SQL are coming from (especially a series of joins), the access control rules
+defined in `skyportal/models.py` for the database model being queried can be a good place to start.
+
+## PostgreSQL
+
+Once you have the full SQL query, it can be easier to spot errors/inefficiencies an to experiment
```
Comment (acrellin) [inline doc/query_analysis.md:68]:
```suggestion
Once you have the full SQL query, it can be easier to spot errors/inefficiencies and to experiment
```

**Review by acrellin — APPROVED** [review]
Nice, thanks @kmshin1397 !

Just a few typos noted below.

**Comment by acrellin** [comment]
@kmshin1397 If there's any more wisdom regarding spotting sub-optimal joins, etc, that'd be great to include (you may have touched all the main points here already though).

**Review by dannygoldstein — APPROVED** [review]

**Pull request merged**